lib/image.ml

```diff
@@ -228,35 +215,155 @@ let parse_src_with_size_info conf s =
       (Format.sprintf "Error parsing portrait source with size info %s" s);
     Error "Failed to parse url with size info"
 
-let get_portrait_with_size conf base p =
+let get_portrait conf base p =
   if has_access_to_portrait conf base p then
     match src_of_string conf (sou base (get_image p)) with
     | `Src_with_size_info _s as s_info -> (
         match parse_src_with_size_info conf s_info with
         | Error _e -> None
-        | Ok (s, size) -> Some (s, Some size))
-    | `Url _s as url -> Some (url, None)
-    | `Path p as path ->
-        if Sys.file_exists p then
-          Some (path, size_from_path path |> Result.to_option)
-        else None
+        | Ok (s, _size) -> Some s)
+    | `Url _s as url -> Some url
+    | `Path p as path -> if Sys.file_exists p then Some path else None
     | `Empty -> (
         match full_portrait_path conf base p with
         | None -> None
-        | Some path -> Some (path, size_from_path path |> Result.to_option))
+        | Some path -> Some path)
   else None
 
-let get_portrait conf base p =
+(* In images/keydir we store either
+   - the old portrait as the original keydir.jpg/png/tif image
+   - the url to the portrait as content of a url.txt file
+*)
+let get_old_portrait conf base p =
+  let key = default_portrait_filename base p in
+  let p_dir =
+    String.concat Filename.dir_sep [ Util.base_path [ "images" ] conf.bname ]
+  in
+  let f = Filename.concat (Filename.concat p_dir "old") key in
+  (* TODO test for legal extensions *)
+  if Sys.file_exists (f ^ ".jpg") then Some (`Path (f ^ ".jpg"))
+  else if Sys.file_exists (f ^ ".jpeg") then Some (`Path (f ^ ".jpeg"))
+  else if Sys.file_exists (f ^ ".png") then Some (`Path (f ^ ".png"))
+  else if Sys.file_exists (f ^ ".gif") then Some (`Path (f ^ ".gif"))
+  else None
+
+let rename_portrait conf base p (nfn, nsn, noc) =
+  match get_portrait conf base p with
+  | Some (`Path old_f) -> (
+      let new_s = default_portrait_filename_of_key nfn nsn noc in
+      let old_s = default_portrait_filename base p in
+      let f = Filename.concat (Util.base_path [ "images" ] conf.bname) new_s in
+      let new_f = f ^ Filename.extension old_f in
```

Comment:
`Filename.extension old_f` is reused latter, maybe make it a variable

Reply:
yes